near-sdk/src/promise.rs

```diff
@@ -430,6 +430,31 @@ impl borsh::BorshSerialize for Promise {
     }
 }
 
+/// When the method can return either a promise or a value, it can be called with `PromiseOrValue::Promise`
+/// or `PromiseOrValue::Value` to specify which one should be returned.
+///   Example:
+/// ```no_run
+/// # use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+/// # use near_sdk::{ext_contract, near_bindgen, Gas, PromiseOrValue};
+/// #[ext_contract]
+/// pub trait ContractB {
+///     fn b(&mut self);
+/// }
+/// #[near_bindgen]
+/// #[derive(Default, BorshDeserialize, BorshSerialize)]
+/// pub struct ContractA {}
+///
+/// #[near_bindgen]
+/// impl ContractA {
+///     pub fn a(&self, value: Option<bool>) -> PromiseOrValue<bool> {
+///         if let Some(value) = value {
+///             PromiseOrValue::Value(value)
+///         } else {
+///             contract_b::b("bob_near".parse().unwrap(), 0, Gas(1_000)).into()
+///         }
+///     }
+/// }
+/// ```
```

Comment:
Seems more useful if this example is more simple and doesn't contain and of the near_bindgen wrapping.

Something like:
```suggestion
/// # Example
/// ```no_run
/// # use near_sdk::{ext_contract, near_bindgen, Gas, PromiseOrValue};
/// #[ext_contract]
/// pub trait ContractB {
///     fn b(&mut self);
/// }
/// let val: PromiseOrValue<bool> = if let Some(value) = value {
///     PromiseOrValue::Value(value)
/// } else {
///     contract_b::b("bob_near".parse().unwrap(), 0, Gas(1_000)).into()
/// };
/// ```
```

Something like this? I didn't check if it compiles, but just to give an idea